internal/service/cloudprovideraccess/resource_cloud_provider_access_setup.go

```diff
@@ -131,53 +132,41 @@ func resourceMongoDBAtlasCloudProviderAccessSetupRead(ctx context.Context, d *sc
 func resourceMongoDBAtlasCloudProviderAccessSetupCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	projectID := d.Get("project_id").(string)
 
-	conn := meta.(*config.MongoDBClient).Atlas
+	conn := meta.(*config.MongoDBClient).AtlasV2
 
-	requestParameters := &matlas.CloudProviderAccessRoleRequest{
+	requestParameters := &admin.CloudProviderAccessRole{
 		ProviderName: d.Get("provider_name").(string),
 	}
 
 	if value, ok := d.GetOk("azure_config.0.atlas_azure_app_id"); ok {
-		requestParameters.AtlasAzureAppID = conversion.Pointer(value.(string))
+		requestParameters.SetAtlasAzureAppId(value.(string))
 	}
 
 	if value, ok := d.GetOk("azure_config.0.service_principal_id"); ok {
-		requestParameters.AzureServicePrincipalID = conversion.Pointer(value.(string))
+		requestParameters.SetServicePrincipalId(value.(string))
 	}
 
 	if value, ok := d.GetOk("azure_config.0.tenant_id"); ok {
-		requestParameters.AzureTenantID = conversion.Pointer(value.(string))
-	}
-
-	if value, ok := d.GetOk("azure_config.0.atlas_azure_app_id"); ok {
-		requestParameters.AtlasAzureAppID = conversion.Pointer(value.(string))
+		requestParameters.SetTenantId(value.(string))
```

Comment:
if Azure params are not in the API we might want to wait until API is fixed before merging this PR and have breking changes

Reply:
These are actually Azure params, see https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Cloud-Provider-Access/operation/createCloudProviderAccessRole and select AZURE. Does that make sense?

Reply:
ah, ok, but there are some params deleted that are not set now like AtlasAzureAppID or AzureServicePrincipalID?

Reply:
if you look better you'll see they were repeated twice 😄 that's why I removed them. They were set two times

Reply:
ah, ok 🤯

Reply:
@lantoli let me know if I am missing something or if we're good to resolve this :) Thanks